src/models.py

```diff
@@ -196,58 +199,18 @@ class AdvectionDiffusionBGK(LBMBase):
     Advection Diffusion Model based on the BGK model.
     """
 
-    def __init__(self, vel, lattice, omega, nx, ny, nz=0, precision='f32/f32', optimize=False):
-        super().__init__(lattice, omega, nx, ny, nz, precision, optimize=optimize)
-        self.vel = vel
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.vel = kwargs.get("vel", None)
```

Comment:
Should raise an error if vel is not given explicitly

Reply:
Agree